src/core/agent_containers.jl

```diff
@@ -1,21 +1,20 @@
 export SoAType
 
 """
-    ContainerType{A}
-
-Union type representing the possible container types for storing agents.
-"""
-ContainerType{A} = Union{AbstractDict{Int,A},AbstractVector{A}}
-
-"""
-    struct AgentWrapperSoA{C} <: AbstractAgent
+    AgentWrapperSoA <: AbstractAgent
 
 Wrapper type for agents in a StructVector container.
 """
 struct AgentWrapperSoA{A, C} <: AbstractAgent
     soa::C
     id::Int
 end
+
+"""
+`SoAType` is a type alias for `AgentWrapperSoA`.
+
+This alias is provided for convenience and to improve code readability.
+"""
```

Comment:
This will go in the API though which is not relevant for a user.  

```julia
SoAType{A} <: AbstractAgent

Wrapper type for agents of type `A` in a model containing a `StructVector` container.

This is needed for specializing signatures involving `A` e.g. instead of writing `agent_step!(agent::A, model)`,
in the case of a model with a `StructVector` container, you should write `agent_step!(agent::SoaType{A}, model)`
since `model[id]` is a `SoaType{A}`.
```